gusto/io.py

```diff
@@ -310,7 +310,10 @@ def setup_dump(self, state_fields, t, pick_up=False):
 
         if self.output.dump_vtus or self.output.dump_nc:
             # make list of fields to dump
-            self.to_dump = [f for f in state_fields.fields if f.name() in state_fields.to_dump]
+            # FIXME: This is a hack, it shouldn't be necessary to sort this list,
+            # It shouldn't be possible to create the list in different orders on different ranks!!!
+            # Problem: `state_fields.to_dump` is a set()! No guaranteed order
+            self.to_dump = sorted([f for f in state_fields.fields if f.name() in state_fields.to_dump], key=lambda x: x.name())
```

Comment:
Can we resolve this by changing `self.to_dump` where it is created? I think this is line 111 of `fields.py`:
```
class StateFields(Fields):
    ...
    def __init__(self, prognostic_fields, prescribed_fields, *fields_to_dump):
        ...
        self.fields = []
        output_specified = len(fields_to_dump) > 0
        self.to_dump = set((fields_to_dump))
        self.to_pick_up = set(())
```
I don't actually see why this (or `state_fields.to_pick_up` on the line below_ needs to be a set, so can we just change them both to be lists?

Reply:
So the only reason I can think for these being `set()`s rather than `list()`s is to avoid duplication. It might be that some fields are added more than once and duplicates need to be removed. If that's the case you need some sort of sorted set that is consistent over all MPI ranks.